### DISCUSSION TITLE
【Keyball39】左側のLEDが全て点灯しなくなった

### DISCUSSION THREAD

**Discussion by isheik** [Yowkees/keyball#694] [Q&A] [answered]
Keyball39愛用させていただいています。素敵なデバイスをありがとうございます！

左右のキーボード共にLEDが全て点灯する状態でしばらく使用しておりました。ある日気がつくと左側（非トラボ側）のLEDのみ全てつかなくなっていることに気が付きました。左右共にキーボード入力自体は問題なく機能しています。

はんだ付けの見直しや1番目のLEDを交換してみたところ、一度また点灯するようになったのですが、少し使用しているとまた点灯しなくなりました。それ以降稀に点灯することはあるものの、大体非点灯となっている状態です。

はんだ付けのどこかが悪いのかなと考えているのですが、不審な点やPro MicroのどのピンがLEDへの電気入力などに対応しているのかなどご助言いただけますと幸いです。

参考として左側のはんだ付けの状態の写真をいくつか共有させていただきたいと思います。

お手数ですがよろしくお願いいたします。

![PXL_20241101_054023847.jpg](https://github.com/user-attachments/assets/f0859f31-3397-4192-8726-7f47b6bd959a)

![PXL_20241101_054106359.jpg](https://github.com/user-attachments/assets/098ab141-b1d7-427e-b98a-fcbb25af8157)

![PXL_20241101_054031701.jpg](https://github.com/user-attachments/assets/ccfe94d6-6bca-409b-ad3f-cf11a07d5139)

![PXL_20241101_054035304.jpg](https://github.com/user-attachments/assets/d9c64dec-5611-46b5-86a7-46c7c858784e)

![PXL_20241101_054114819.jpg](https://github.com/user-attachments/assets/b92468ae-5747-40d0-aa09-5061071146d8)

![PXL_20241101_054038946.jpg](https://github.com/user-attachments/assets/f6851814-aa5d-4ffd-acda-84b2b43446a1)

**Comment by Yowkees** [comment]
Keyball39をご愛用いただき、誠にありがとうございます！

うーむ。非常にキレイに半田付けできていますし、部品方向等も問題なさそうですね。

LED1を交換後に一時的に復活して、また点灯しなくなったということで、もしかするとLED自体の問題の可能性もあるかと思います。
フルカラーLEDはとても湿気に弱いようで、当方はシリカゲルを入れた袋で保存しています。湿気っていると半田付け時に故障します。
LED1をできるだけ信頼の置けそうなLEDに交換してみると良いかもしれません。

お手数ですが、ご確認お願いいたします。

**Reply by isheik** [reply]
ご返信ありがとうございます！

>うーむ。非常にキレイに半田付けできていますし、部品方向等も問題なさそうですね。

半田付けお褒めいただきありがとうございます。Keyball39をきっかけに人生初の半田付けに挑戦したためとても嬉しいです！

LEDは湿気にも弱いのですね。再度LED交換を検討してみたいと思います。LED交換の際、念のため一緒にLED1に関連するPro Micro側の半田付けも見直したいと考えております。お手数ですがLED1に関連するPro Microのピンをご教示いただくことは可能でしょうか。

公開いただいている下記Keyball39 (非トラボ側) の回路図を参照しているのですが、回路には明るくなく正しく読めているかあまり自信が持てずにおります。
https://github.com/Yowkees/keyball/blob/be5ef1009116af22533c0b8b72ae2fb7db9b8610/keyball39/design_data/Circuit_of_Keyball39_NoBallside_rev2.pdf

下記の写真でLED(TXO)、VCCと記載されたピンがそれぞれLED1への電源供給または制御信号伝達を担っているピンという認識でよろしかったでしょうか。
![Screenshot 2024-11-21 at 10 45 58 PM](https://github.com/user-attachments/assets/7e3820c7-6896-40cc-aefd-55c543e4cc49)

お手数ですがよろしくお願いいたします。

**Comment by Yowkees** [comment]
はい。おっしゃる通り、LEDと表記のあるピンです。一番上ですね。
あとVCCとGNDが通電していれば点灯するはずです。

お手数ですがご確認よろしくお願いいたします。

**Reply by isheik** [reply]
ご返信ありがとうございます。はんだ付け見直しやLEDの再交換の前に少し試行錯誤してみたのですが、Pro MicroのUSB-CコネクタかPro Micro自体が怪しいかもしれないということがわかりました。

というのも、Pro Microに挿したUSB-Cケーブルをコネクタに押し込むとLEDが正常に動作することがあったためです。以下YouTubeに参考として挙動の動画をあげさせていただいております。
https://www.youtube.com/shorts/82AODfnlb30

動作している時はLEDは想定した通り色変更なども行われ挙動に問題はございません。ただ押し込みを離して少しすると色変更に正しく反応せずLEDがフリーズした状態になります。

USB-Cコネクタ関連の不備と考えた時、解せない点として以下2点がございます。

1. 現在USB-Cにはマグネットコネクタを使用していてこちらの接触不備かもしれない……とも思ったのですがマグネットコネクタを使用せずに直接ケーブルを挿すと、検証した範囲では逆に何をしても全くLEDがつかない状態になる。（マグネットコネクタがたまたまUSB-Cコネクタの接触を良い状態にしている？）

2. 右側（問題のないトラボ側のキーボード）を親としてUSBケーブルをさした場合、問題の左側のLEDは全くつかなくなる。（USB-Cコネクタが関係した給電の問題ならTRRSケーブル経由で給電されている時は正常に動くはず）

これらの状況からLEDやPro MicroのUSB-Cコネクタよりも、問題のある左側キーボードのPro Micro自体や基盤内での送電の問題の可能性が高いと考えております。また全てのケースにおいて問題が発生しているのは左側キーボードのLEDのみで、押されたキーの認識は問題ございません。

上記状況を見て、もしYowkees様の観点から他に怪しいと思われる点がございましたらお教えいただけますと幸いです。よろしくお願いいたします。

**Reply by isheik** [reply]
もう少しリサーチしてみたところ、キーボードは異なりますがRedditの投稿でほぼ同じような症状を見つけました。
こちらの解決例を見るとやはりPro Microのはんだ付けがうまくいっていないかPro Microの部分的な不良の可能性が高そうなのでそちらを見直してみようと思います。ありがとうございます🙇

https://www.reddit.com/r/crkbd/s/FVCtu6X8PP

**Reply by isheik** [reply, marked as answer]
生活がバタバタしており時間がとれず見直しに少し間があいてしまいました。すみません。
本日半田付けの見直しを行ってみたのですが、無事に直すことができました！

今回以下のことを行いました:
1. 左側のPro Microの裏側の全てのピンの半田を再加熱して再定着させる (新たな半田を足すことはせず
2. OLEDのアクリル板の取り付けネジを少し緩くする

1の半田付け見直しを完了した時点のLEDテストでは直らず。その後Pro Microの表側のピンの半田を全て再加熱するか検討するためとりあえずOLEDのアクリル板を外しました。その際念のため再度LEDテストをしてみたところLEDが正常に動くようになりました。OLEDのアクリル板の取り付けネジを締め付けすぎていたのかなと思い、気持ち前より緩めたところ安定してLEDが稼働するようになりました。

上記1と2のどちら、または両方のおかげで現象が改善したのかいまいちはっきりわかりませんが、類似の症状の方がいらっしゃるかもしれないので記録として共有させていただきます。

ご助言いただきありがとうございました！これからも楽しくKeyball39を使わせていただきたいと思います。本当に素敵なデバイスをお作りいただきありがとうございます！
